project-koku/koku
# User settings endpoints

Additional settings endpoints that will return the users preferred settings in a simple format without data driven forms

Testing Instructions:
1. pull and checkout branch
2. Test the following 3 endpoints:

Users settings endpoint which returns all user preferred settings without tags
```
http://127.0.0.1:8000/api/cost-management/v1/account-settings/

Feedback should look as follows:
"meta": {
        "count": 1
    },
    "links": {
        "first": "/api/cost-management/v1/user-settings/?limit=10&offset=0",
        "next": null,
        "previous": null,
        "last": "/api/cost-management/v1/user-settings/?limit=10&offset=0"
    },
    "data": {
            "currency": "GBP",
            "cost_type": "savingsplan_effective_cost"
    }
```

user settings cost-type endpoint which returns the users preferred cost-type only

```
http://127.0.0.1:8000/api/cost-management/v1/account-settings/cost-type/

Feedback should look as follows:
"meta": {
        "count": 1
    },
    "links": {
        "first": "/api/cost-management/v1/user-settings/cost-type/?limit=10&offset=0",
        "next": null,
        "previous": null,
        "last": "/api/cost-management/v1/user-settings/cost-type/?limit=10&offset=0"
    },
    "data": {
        "cost_type": "savingsplan_effective_cost"
    }

```

user settings currency endpoint which returns the users preferred currency only

```
http://127.0.0.1:8000/api/cost-management/v1/account-settings/currency/

Feedback should look as follows:
"meta": {
        "count": 1
    },
    "links": {
        "first": "/api/cost-management/v1/user-settings/currency/?limit=10&offset=0",
        "next": null,
        "previous": null,
        "last": "/api/cost-management/v1/user-settings/currency/?limit=10&offset=0"
    },
    "data": {
            "currency": "GBP"
    }
```
